tests/backend/user/test_human_find.py

```diff
@@ -220,77 +229,167 @@ async def test_pagination(access_testbed, local_device_factory):
+            ),
+            HumanFindResultItem(
+                user_id=blacky4.user_id, human_handle=blacky4.human_handle, revoked=False
+            ),
         ],
-        "per_page": 4,
-        "page": 1,
-        "total": 8,
-    }
+        per_page=4,
+        page=1,
+        total=8,
+    )
 
     # Continue pagination
     rep = await human_find(sock, page=2, per_page=4)
-    assert rep == {
-        "status": "ok",
-        "results": [
-            {"user_id": godfrey1.user_id, "human_handle": godfrey1.human_handle, "revoked": False},
-            {"user_id": mike.user_id, "human_handle": mike.human_handle, "revoked": False},
-            {"user_id": richard.user_id, "human_handle": richard.human_handle, "revoked": False},
-            {"user_id": roger.user_id, "human_handle": roger.human_handle, "revoked": False},
+    assert rep == HumanFindRepOk(
+        results=[
+            HumanFindResultItem(
+                user_id=godfrey1.user_id, human_handle=godfrey1.human_handle, revoked=False
+            ),
+            HumanFindResultItem(
+                user_id=mike.user_id, human_handle=mike.human_handle, revoked=False
+            ),
+            HumanFindResultItem(
+                user_id=richard.user_id, human_handle=richard.human_handle, revoked=False
+            ),
+            HumanFindResultItem(
+                user_id=roger.user_id, human_handle=roger.human_handle, revoked=False
+            ),
         ],
-        "per_page": 4,
-        "page": 2,
-        "total": 8,
-    }
+        per_page=4,
+        page=2,
+        total=8,
+    )
 
     # Test out of pagination
     rep = await human_find(sock, page=3, per_page=4)
-    assert rep == {"status": "ok", "results": [], "per_page": 4, "page": 3, "total": 8}
+    assert rep == HumanFindRepOk(results=[], per_page=4, page=3, total=8)
 
     # Test sort is before pagination when pagination and test non-sensitive sort
     rep = await human_find(sock, page=1, per_page=1, query="BlaCkY")
-    assert rep == {
-        "status": "ok",
-        "results": [
-            {"user_id": blacky.user_id, "human_handle": blacky.human_handle, "revoked": False}
+    assert rep == HumanFindRepOk(
+        results=[
+            HumanFindResultItem(
+                user_id=blacky.user_id, human_handle=blacky.human_handle, revoked=False
+            )
         ],
-        "per_page": 1,
-        "page": 1,
-        "total": 4,
-    }
+        per_page=1,
+        page=1,
+        total=4,
+    )
 
 
 @pytest.mark.trio
 async def test_bad_args(access_testbed, local_device_factory):
     binder, org, godfrey1, sock = access_testbed
+    cmd = human_find
     # Test bad params
-    for bad in [{"page": 0}, {"per_page": 0}, {"per_page": 101}]:
-        rep = await human_find(sock, **bad)
-        assert rep["status"] == "bad_message"
+    # We should not be able to build an invalid request
+    for raw_req in [
+        # Generated from Python implementation (Parsec v2.11.1+dev)
```

Comment:
Do we have also those test cases in `libparsec/crates/protocol/tests/tests_user.rs` ? (I think it's fine for the moment to have this duplication, and in the long run we should have only this test in the rust protocol tests)

Reply:
Not the same tests, but added unit tests